tools/wasm-sourcemap.py

```diff
@@ -142,173 +130,74 @@ def get_code_section_offset(wasm):
-      if collect_sources:
-        load_name = prefixes.load.resolve(file_name)
-        try:
-          with open(load_name, 'r') as infile:
-            source_content = infile.read()
-          sources_content.append(source_content)
-        except:
-          print('Failed to read source: %s' % load_name)
-          sources_content.append(None)
-    else:
-      source_id = sources_map[source_name]
-
-    address_delta = address - last_address
-    source_id_delta = source_id - last_source_id
-    line_delta = line - last_line
-    column_delta = column - last_column
-    mappings.append(encode_vlq(address_delta) + encode_vlq(source_id_delta) + encode_vlq(line_delta) + encode_vlq(column_delta))
-    last_address = address
-    last_source_id = source_id
-    last_line = line
-    last_column = column
-  return OrderedDict([('version', 3),
-                      ('names', []),
-                      ('sources', sources),
-                      ('sourcesContent', sources_content),
-                      ('mappings', ','.join(mappings))])
+    sources[source_id] = source_name
+    if collect_sources:
+      load_name = prefixes.load.resolve(file_name)
+      try:
+        with open(load_name, 'r') as infile:
+          source_content = infile.read()
+        sources_content.append(source_content)
+      except:
+        print('Failed to read source: %s' % load_name)
+        sources_content.append(None)
+  map['sourcesContent'] = sources_content
+
+def dwarf_to_json(wasm):
+  logging.debug('Read %s and convert to json' % wasm)
+
+  js = """const fs = require("fs"); const path = require("path");
+const __scriptdir = path.dirname(process.argv[1]);
+async function convertDwarfToJSON(input, enableXScopes = false) {
+    const dwarf_to_json_code = fs.readFileSync(__scriptdir + "/dwarf_to_json.wasm");
+    const {instance: {exports: { alloc_mem, free_mem, convert_dwarf, memory }}} =
+        await WebAssembly.instantiate(dwarf_to_json_code);
+    wasm = fs.readFileSync(input)
+    const wasmPtr = alloc_mem(wasm.byteLength);
+    new Uint8Array(memory.buffer, wasmPtr, wasm.byteLength).set(new Uint8Array(wasm));
+    const resultPtr = alloc_mem(12);
+    convert_dwarf(wasmPtr, wasm.byteLength, resultPtr, resultPtr + 4, enableXScopes);
+    free_mem(wasmPtr);
+    const resultView = new DataView(memory.buffer, resultPtr, 12);
+    const outputPtr = resultView.getUint32(0, true), outputLen = resultView.getUint32(4, true);
+    free_mem(resultPtr);
+    const output = Buffer.from(new Uint8Array(memory.buffer, outputPtr, outputLen)).toString("utf8");
+    free_mem(outputPtr);
+    return output;
+}
+convertDwarfToJSON(process.argv[2], false).then(json => console.log(json));"""
+
+  process = Popen(["node", "-e", js, "--", sys.argv[0], wasm], stdout=PIPE)
+  output, err = process.communicate()
+  exit_code = process.wait()
+  if exit_code != 0:
+    logging.error('Error during dwarf-to-json execution (%s)' % exit_code)
+    sys.exit(1)
```

Comment:
I would replace these six lines with just `output = subprocess.check_output(...)`.  It returns the stdout and throws if non-zero exit code.